pkg/providers/k8s/endpoint/base.go

```diff
@@ -131,3 +131,23 @@ func (c *baseEndpointController) syncEndpoint(ctx context.Context, ep kube.Endpo
 	}
 	return nil
 }
+
+func (c *baseEndpointController) syncEmptyEndpoint(ctx context.Context, ep kube.Endpoint) error {
+	namespace, err := ep.Namespace()
+	if err != nil {
+		return err
+	}
+	svcName := ep.ServiceName()
+	log.Infow("The syncEndpoint, service has been deleted, try to update upstream",
+		zap.String("Namespace", namespace),
+		zap.String("ServiceName", svcName),
+	)
+	clusterName := c.Config.APISIX.DefaultClusterName
+	err = c.APISIX.Cluster(clusterName).UpstreamServiceRelation().Delete(ctx, namespace+"_"+svcName)
+	if err != nil {
+		log.Errorw("try to delete upstream relation and upstream nodes failed",
```

Comment:
```suggestion
		log.Errorw("delete upstream relation failed",
```

While this function does delete nodes, the log doesn't need to display it.